### PR TITLE
NMS-16950: break up jdbc-datacollection-config.xml

### PR DESCRIPTION
**Pull request by dino2gnt**
`jdbc-datacollection-config.xml` -> `jdbc-datacollection.d/{postgres.xml|mysql.xml}`  and a smaller `jdbc-datacollection-config.xml`

This makes the jdbc datacollection config consistent with the majority of the other collectors configs.

Doing this broke two tests in such a way that I was unable to figure out how to fix them. There were no examples I could find in the other collectors that use the abstract merging config dao for a similar test to base my work on, and I'm not enough of a software engineer to figure it out from scratch.  Happy to take an assist if having working tests for this feature is important; the lack of other tests in other collectors I could copy from leads me to believe it's not as important as some might think :)

I also fixed a bug with the jdbc-datacollection xsd that caused some valid configurations to fail validation. (resource-type is actually _not_ required)

This has been function tested in a local 33.0.6 instance, so I don't think anything should blow up :rofl: 

### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://opennms.atlassian.net/) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the Jira link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the Jira issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-16950

### CODE REVIEW

File: features/jdbc-collector/src/main/java/org/opennms/netmgt/config/jdbc/JdbcDataCollectionConfig.java [hunk 1]
```diff
@@ -97,7 +97,28 @@ public JdbcDataCollection getDataCollectionByName(String name) {
...
+        if (other == null) {
+            return this;
+        }
+        // Overwrite the rrdRepository iff it's null
+        if (m_rrdRepository == null && other.m_rrdRepository != null) {
+            m_rrdRepository = other.m_rrdRepository;
+        }
+        // Merge the lists of collections and groups
+        getCollection().addAll(other.getCollection());
+        //getGroup().addAll(other.getGroup());
```
Comment (fooker) [inline features/jdbc-collector/src/main/java/org/opennms/netmgt/config/jdbc/JdbcDataCollectionConfig.java:111]: No groups for JDCB? If not, remove this line.

**Review by fooker — CHANGES_REQUESTED** [review]
Overall, looking good. Can we re-enable the tests?

File: features/jdbc-collector/src/main/java/org/opennms/netmgt/config/jdbc/JdbcDataCollectionConfig.java (hunk 1 above)
Comment (dino2gnt) [reply features/jdbc-collector/src/main/java/org/opennms/netmgt/config/jdbc/JdbcDataCollectionConfig.java]: Nope, no groups.  I've removed the commented line.

**Review by fooker — APPROVED** [review]

**Comment by dino2gnt** [comment]
Adding the tests back broke compilation, and I don't know the best way to make the tests compile after these changes.  What is the path forward?

**Review by cgorantla — COMMENTED** [review]
These changes looks good to me.

You can remove JdbcDataCollectionConfigDaoJaxbTest.java , this test is no longer valid.

**Review by cgorantla — APPROVED** [review]
LGTM!

**Review by indigo423 — APPROVED** [review]
LGTM! Great improvement, thank you!

**Pull request merged**